e2e/fiat_tf_test.go

```diff
@@ -0,0 +1,612 @@
+	require.NoError(t, err, "error configuring minter controller")
+
+	showMCRes, err := showMinterController(ctx, val, minterController1)
+	require.NoError(t, err, "failed to query show-minter-controller")
+	expectedShowMinterController := fiattokenfactorytypes.QueryGetMinterControllerResponse{
+		MinterController: fiattokenfactorytypes.MinterController{
+			Minter:     minter1.FormattedAddress(),
+			Controller: minterController1.FormattedAddress(),
+		},
+	}
+
+	require.Equal(t, expectedShowMinterController.MinterController, showMCRes.MinterController)
+
+	pauseFiatTF(t, ctx, val, nw.fiatTfRoles.Pauser)
+
+	allowance := int64(10)
+
+	_, err = val.ExecTx(ctx, minterController1.KeyName(), "fiat-tokenfactory", "configure-minter", minter1.FormattedAddress(), fmt.Sprintf("%duusdc", allowance))
+	require.ErrorContains(t, err, "minting is paused")
+
+	_, err = showMinters(ctx, val, minter1)
+	require.Error(t, err, "minter found; configuring minter should not have succeeded")
+
+	unpauseFiatTF(t, ctx, val, nw.fiatTfRoles.Pauser)
+
+	// ACTION: Configure minter from a minter controller not associated with the minter
+	// EXPECTED: Request fails; Minter is not configured with new Minter Controller but old Minter retains its allowance
+	// Status:
+	// 	minterController1 -> minter1 (un-configured)
+	// 	gw minterController -> gw minter
+
+	// reconfigure minter1 to ensure balance does not change
+	configureMinter(t, ctx, val, minterController1, minter1, allowance)
+
+	// reconfigure minter1 with a new allownace from a minter controller not associated with the minter
+	differentAllowance := allowance + 99
+	_, err = val.ExecTx(ctx, nw.fiatTfRoles.MinterController.KeyName(), "fiat-tokenfactory", "configure-minter", minter1.FormattedAddress(), fmt.Sprintf("%duusdc", differentAllowance))
+	require.ErrorContains(t, err, "minter address ≠ minter controller's minter address")
+
+	showMintersRes, err := showMinters(ctx, val, minter1)
+	require.NoError(t, err, "failed to query show-minters")
+	expectedShowMinters := fiattokenfactorytypes.QueryGetMintersResponse{
+		Minters: fiattokenfactorytypes.Minters{
+			Address: minter1.FormattedAddress(),
+			Allowance: sdktypes.Coin{
+				Denom:  "uusdc",
+				Amount: math.NewInt(allowance),
+			},
+		},
+	}
+	require.Equal(t, expectedShowMinters.Minters, showMintersRes.Minters, "configured minter allowance is not as expected")
+
+	// ACTION: Configure a minter is blacklisted from a blacklisted minter controller
+	// EXPECTED: Success; Minter is configured with allowance
+	// Status:
+	// minterController1 -> minter1
+	// gw minterController -> gw minter
+
+	blacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, minterController1)
+
+	blacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, minter1)
+
+	configureMinter(t, ctx, val, minterController1, minter1, 11)
```

Comment:
missing, verify the minter was configured and the allowance is set properly.

Reply:
See: https://github.com/noble-assets/noble/pull/368/files#r1719073950